src/Pos/Launcher/Resource.hs

```diff
@@ -202,12 +204,32 @@ bracketNodeResources :: forall ssc m a.
     -> SscParams ssc
     -> (NodeResources ssc m -> Production a)
     -> Production a
-bracketNodeResources np sp k = bracketTransport tcpAddr $ \transport ->
+bracketNodeResources np sp k =
+    bracketTransport tcpAddr $ \transport ->
     bracketKademlia (npBaseParams np) (npNetworkConfig np) $ \networkConfig ->
-        bracket (allocateNodeResources transport networkConfig np sp) releaseNodeResources k
+    bracket (allocateNodeResources transport networkConfig np sp) releaseNodeResources $ \nodeRes -> do
+      case npInitDelay np of
+        Nothing ->
+          -- Start node immediately
+          return ()
+        Just initDelay -> do
+          now <- liftIO $ Time.getCurrentTime
+          let delay = (toUSec . toSec) (initDelay `Time.diffUTCTime` now)
+          if delay >= 0 then do
+            logNotice $ sformat ("Delaying node startup until " % shown) initDelay
+            liftIO $ threadDelay delay
+          else do
+            logWarning $ sformat ("--init-delay parameter in the past")
```

Comment:
Move all of this initDelay stuff into a function in a where clause, so the top level bit reads more clearly, e.g. as just:

      bracket (allocateNodeResources transport networkConfig np sp) releaseNodeResources $ \nodeRes -> do
        waitForStartTime (npInitDelay np)
        k nodeRes
      where
        waitForStartTime Nothing = ...
        waitForStartTime (Just initDelay) = ...